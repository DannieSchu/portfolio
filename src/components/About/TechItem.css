.TechItem {
  width: 100%;
  display: flex;
  margin: 6%; 
  
  dt.title {
    width: 40%;
  }  

  div.definitions {
    display: flex;
    flex-direction: column;
    width: 40%;
    dd {
      font-weight: 300;
      -webkit-margin-before: 0;
      -webkit-margin-after: 0;
    }
  }

  @media only screen and (max-width: 768px) {
    margin: 5%; 
  }
}
