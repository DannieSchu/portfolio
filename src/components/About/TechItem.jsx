import React from 'react';
import PropTypes from 'prop-types';

const TechItem = ({ title, definitions }) => {
  const definitionElements = definitions.map((definition, i) => <dd key={i}>{definition}</dd>);

  return (
    <dl>
      <dt>{title}</dt>
      {definitionElements}
    </dl>
  );
};

TechItem.propTypes = {
  title: PropTypes.string.isRequired,
  definitions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default TechItem;

