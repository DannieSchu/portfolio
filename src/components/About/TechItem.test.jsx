import React from 'react';
import { shallow } from 'enzyme';
import TechItem from './TechItem';

describe('TechItem component', () => {
  it('renders TechItem', () => {
    const wrapper = shallow(<TechItem title="Languages" definitions={['JavaScript', 'HTML', 'CSS']}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
