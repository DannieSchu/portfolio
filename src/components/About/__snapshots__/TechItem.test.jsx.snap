// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechItem component renders TechItem 1`] = `
<article
  className="TechItem"
>
  <dt
    className="title pullOut"
  >
    Languages
  </dt>
  <div
    className="definitions"
  >
    <dd
      className="pullOut"
      key="0"
    >
      JavaScript
    </dd>
    <dd
      className="pullOut"
      key="1"
    >
      HTML
    </dd>
    <dd
      className="pullOut"
      key="2"
    >
      CSS
    </dd>
  </div>
</article>
`;
